### PR TITLE
Add authentication for websocket feed

### PR DESCRIPTION
**Pull request by CjS77**
This will allow users to receive user-specific messages on the websocket feed in future

### CODE REVIEW

File: lib/clients/authenticated.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ var util = require('util');
 var crypto = require('crypto');
```
Comment (mihar) [inline lib/clients/authenticated.js]: Remove dependency if unused

File: lib/clients/authenticated.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ var util = require('util');
...
 var querystring = require('querystring');
```
Comment (mihar) [inline lib/clients/authenticated.js]: Remove dependency if unused

File: lib/clients/websocket.js [hunk 3]
```diff
@@ -2,11 +2,23 @@ var EventEmitter = require('events').EventEmitter;
 var Websocket = require('ws');
 var util = require('util');
 var _  = {assign: require('lodash.assign')};
-
-var WebsocketClient = function(productID, websocketURI) {
+var crypto = require('crypto');
```
Comment (mihar) [inline lib/clients/websocket.js]: Remove dependency if unused

File: lib/clients/websocket.js [hunk 4]
```diff
@@ -67,6 +85,9 @@ _.assign(WebsocketClient.prototype, new function() {
     var self = this;
     self.emit('message', JSON.parse(data));
   };
-});
 
+  prototype.getSignature = function(auth) {
```
Comment (mihar) [inline lib/clients/websocket.js]: Don't need to pass `auth` since it's on the object

File: lib/clients/websocket.js [hunk 5]
```diff
@@ -43,14 +55,20 @@ _.assign(WebsocketClient.prototype, new function() {
   prototype.onOpen = function() {
     var self = this;
     self.emit('open');
+    var auth = self.getSignature(self.auth);
```
Comment (mihar) [inline lib/clients/websocket.js]: Variable should be called `sig`

File: lib/clients/websocket.js (hunk 4 above)
Comment (CjS77) [inline lib/clients/websocket.js]: The 1-liner is actually redundant, so it's removed

**Pull request merged**